cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/RepositoryManager.java

```diff
@@ -0,0 +1,201 @@
+    this.secureStore = secureStore;
+    this.sourceControlConfig = sourceControlConfig;
+  }
+
+  /**
+   * Returns the base path in the git repo to store CDAP applications. If an optional Path prefix is provided in the
+   * repository configuration, the returned path includes it.
+   *
+   * @return the path for the repository base directory.
+   */
+  public Path getBasePath() {
+    Path localRepoPath = sourceControlConfig.getLocalRepoPath();
+    String pathPrefix = sourceControlConfig.getRepositoryConfig().getPathPrefix();
+    if (pathPrefix == null) {
+      return localRepoPath;
+    }
+    return localRepoPath.resolve(pathPrefix);
+  }
+
+  /**
+   * Validates the provided configuration.
+   *
+   * @param secureStore         A secure store for fetching credentials if required.
+   * @param sourceControlConfig Configuration for source control operations.
+   * @throws RepositoryConfigValidationException when provided repository configuration is invalid.
+   */
+  public static void validateConfig(SecureStore secureStore, SourceControlConfig sourceControlConfig) throws
+    RepositoryConfigValidationException {
+    RepositoryConfig config = sourceControlConfig.getRepositoryConfig();
+    // Try fetching heads in the remote repository.
+    InMemoryRepository localRepo;
+    localRepo = new InMemoryRepository(new DfsRepositoryDescription());
+    try (Git git = new Git(localRepo)) {
+      LsRemoteCommand cmd = createCommand(git::lsRemote, sourceControlConfig, secureStore).setRemote(config.getLink())
+                                                                                          .setHeads(true)
+                                                                                          .setTags(false);
+      validateDefaultBranch(cmd.callAsMap(), config.getDefaultBranch());
+    } catch (TransportException e) {
+      throw new RepositoryConfigValidationException(
+        "Failed to connect with repository. Authentication credentials invalid or the repository may not exist.",
+        e);
+    } catch (RepositoryConfigValidationException e) {
+      throw e;
+    } catch (AuthenticationConfigException e) {
+      throw new RepositoryConfigValidationException(e.getMessage(), e);
+    } catch (Exception e) {
+      throw new RepositoryConfigValidationException("Failed to list remotes in remote repository.", e);
+    }
+  }
+
+  /**
+   * Initializes the Git repository by cloning remote.
+   *
+   * @throws GitAPIException               when a Git operation fails.
+   * @throws AuthenticationConfigException when correct authentication credentials are not available.
+   * @throws IOException                   when file or network I/O fails.
+   */
+  @Override
+  protected void startUp() throws Exception {
+    if (git != null) {
+      return;
+    }
+    // Clean up the directory if it already exists.
+    deletePathIfExists(sourceControlConfig.getLocalRepoPath());
```

Comment:
We actually might not want to delete the path if it is a valid repo from previous invocation. This can be a future improvement.

Reply:
Created https://cdap.atlassian.net/browse/CDAP-20341 for improvement. Will raise a PR for that soon.